ufp: add a device fingerprinting daemon

blogic opened the pull request:
The two daemons in this PR allow users to fingerprint devices on their network.

The final fingerprints look like this:

```
    {
            "04:99:b9:XX:YY:ZZ": {
                    "device": "HomePod mini",
                    "vendor": "Apple",
                    "class": "speaker",
                    "device_name": "Kitchen"
            },
            "c4:65:16:XX:YY:ZZ": {
                    "vendor": "HP",
                    "device": "OfficeJet Pro 9010 series",
                    "class": "Printer"
            }
    }
```

and can be queried via an ubus API. Further details are explained in the patches.

Signed-off-by: John Crispin <john@phrozen.org>

nbd168 committed:
ramips: add work-in-progress support for D-Link DIR-X1860

Signed-off-by: Felix Fietkau <nbd@nbd.name>

blogic committed:
ramips: add support for actiontec web7200

Signed-off-by: John Crispin <john@phrozen.org>

nbd168 committed:
gcc/binutils: disable sysv style hash tables for generated binaries

GNU hash tables are always preferred by musl, so the sysv ones are completely
unused and only waste space. Reduces image size by ~1.4%

Signed-off-by: Felix Fietkau <nbd@nbd.name>

blogic committed:
base-files: add support for the pattern trigger to the led script

https://www.kernel.org/doc/Documentation/devicetree/bindings/leds/leds-trigger-pattern.txt

Signed-off-by: John Crispin <john@phrozen.org>

nbd168 committed:
hostapd: add AFC support

Signed-off-by: Felix Fietkau <nbd@nbd.name>

KanjiMonster commented:
Does this require functionality from main, or would this also work on top of a 23.05 image?

blogic commented:
@KanjiMonster requires stuff from main. mainly the hostapd ubus API to get the IEs used for taxonomy

dhewg commented:
I already mentioned some points on irc, I'll add it here before it's too 
scattered:
- The package feeds already has a package of the same name which also 
registers an ubus object of the same name. If you pulled that package in 
your local build, the one added here won't be used
- "//usr/lib/opkg/info/udhcpsnoop.postinst: 
/etc/uci-defaults/50-dhcpsnoop: line 13: lan1: not found"
   looks like there's a "lan=" missing in front of the cat
- `ubus call dhcpsnoop dump` shows an entry with an ip of 0.0.0.0, but 
`arp -a` shows that mac has a valid lan ip

tapper82 commented:
Hi will there be a luci app for device fingerprinting? Would be good for me to see what's on my network.

colinw-smartrg commented:
Are you married to to the naming of the fields device and device_name ?
The term "device" is overused and not very informative in the structure. 
It appears the field device is the Model or Description of the device.
The term "device_name" appears to be the hostname from DHCP. Shouldn't we use the term "hostname" from the DHCP RFC ?

blogic committed:
hostapd: add ubus support for wired driver

Signed-off-by: John Crispin <john@phrozen.org>

blogic committed:
mediatek: use wpad-mbedtls instead of wpad-basic-mbedtls

Signed-off-by: John Crispin <john@phrozen.org>

nbd168 committed:
mac80211: add WIFI_SCRIPTS_UCODE to PKG_CONFIG_DEPENDS

It affects package depdendencies

Signed-off-by: Felix Fietkau <nbd@nbd.name>

Neustradamus commented:
@blogic: Have you seen the latest comments?

blogic committed:
procd: procd_/open/add/close/_firewall() helper functions

Signed-off-by: John Crispin <john@phrozen.org>

blogic committed:
umdns: use new procd helper functions when registering firewall ports

Signed-off-by: John Crispin <john@phrozen.org>

blogic committed:
hostapd: improve built-in radius server handling

* generate the clients file on demand
* allow adding interfaces to the procd data section

Signed-off-by: John Crispin <john@phrozen.org>

blogic committed:
base-files: rc.button: improve button handling

Move scripts to /etc/rc.button.d/ and add a ucode script that invoke the
button handler. The script allows adding/overriding button handlers
via UCI. A sample /etc/config/system section look like this

config button
        option button wps
        option script /usr/libexec/wps-button

Signed-off-by: John Crispin <john@phrozen.org>

blogic committed:
base-files: add -m to sysupgrade

This behaves the same way as putting rootfs_data_max into the uboot-env.
This value is used to restrict the size of rootfs_data on NAND flash.
The free spaec can be used by uvol.

Signed-off-by: John Crispin <john@phrozen.org>

blogic committed:
rc.button

PolynomialDivision commented:
What is the actual status of this PR? Why has there been no progress for several months?

nbd168 committed:
netifd: update to Git mlo (2025-04-02)

3b100d242223 CMakeLists.txt: bump minimum required version
a213d30b1856 scripts: fix dummy mode on systems where libubox is in /usr/local
0c96e7aab6ac update example mac80211 script and wireless config
43cadb39b5e7 wireless: replace with ucode scripts

Signed-off-by: Felix Fietkau <nbd@nbd.name>

ckorber commented:
> What is the actual status of this PR? Why has there been no progress for several months?

I would like to jump on board and ask whether any changes are planned for this PR in the near future.

I have written a Python script based on it that calls https://standards-oui.ieee.org/ and generates a json file for each OUI in the same way as ufp does it, in order to add it to devices.bin.

This way every vendor name of devices in the network could be recognized by ufp.

blogic commented:
where does the python script run ? on the AP or do you use it to generate an oui binary that gets used by the AP for lookup

ckorber commented:
Currently it runs a private repo where it generates json files. I will have to move it to a public one.
I have overriden `Download` in the ufp Makefile to download and extract it into `data`. This way it is compiled into the existing `devices.bin`.

ckorber commented:
For every entry it is 2.2MB currently.

blogic commented:
are there multiple entries ?

ckorber commented:
Ultimately, these are all entries from the website https://standards-oui.ieee.org/.
I had alternatives paired by vendor and such, but that didn't do much to the size.

blogic commented:
sounds reasonable, it should be a menuconfig option where the user can choose to include the oui db or not depending on the devices space

ckorber commented:
Thats a good idea.
I will move the repo tomorrow and add this. When I'm done I will add it to your repo as a PR.

nbd168 committed:
unetmsg: add support for sending requests/messages to specific hosts

Makes it possible to implement unetmsg APIs for host control

Signed-off-by: Felix Fietkau <nbd@nbd.name>

nbd168 committed:
unetmsg: fix host name in response messages

Avoid using loop iterator variable when generating the closure

Signed-off-by: Felix Fietkau <nbd@nbd.name>

blogic committed:
hostapd: fix sae_password_file support

Signed-off-by: John Crispin <john@phrozen.org>

blogic commented:
I have been reworking the package and will create a PR in the packages repo shortly
